### PR TITLE
Refactor SpecFile._process_patches()

### PR DESCRIPTION
**Pull request by nforro**
Make it public and usable on its own, not only from within `SpecFile.write_updated_patches()`.

### CODE REVIEW

**Comment by nforro** [comment]
Related to https://github.com/packit/packit/issues/1081.

File: rebasehelper/specfile.py [hunk 1]
```diff
@@ -455,7 +460,8 @@ def is_empty(line):
             if patch_removed:
                 # remove the line of the patch that was removed
                 self.removed_patches.append(patch_name)
-                removed_patches.append(tag.index)
+                if tag.index:
```
Comment (FrNecas) [inline rebasehelper/specfile.py:463]: Are these added `if`s necessary? As far as I am aware, only patches should get up to this point and they should always have an index, shouldn't they?

Comment (nforro) [reply rebasehelper/specfile.py:463]: Of course, but the type is still `Optional[int]`.

**Review by FrNecas — APPROVED** [review]

**Review by FrNecas — APPROVED** [review]
Re-approving with the public API test :)

**Pull request merged**